### ISSUE TITLE
[DownloadBuildArtifactsV0] Should support no_proxy

### ISSUE THREAD

**Issue by hangy** [microsoft/azure-pipelines-tasks#16286]
**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: DownloadBuildArtifacts@0

## Environment
- Server: Azure DevOps Server 2020 Update 1.1
- Agent: VstsAgent 2.202.0 on Windows Server 2019

## Issue Description

Currently, our (old style) release pipelines are blocked, because the DownloadBuildArtifactsV0 task does not understand the `no_proxy` environment variable, and tries to download build artifacts from the AzDO server, which is not available through the corporate proxy. The AzDO server is only available internally, and thus listed in `no_proxy`.

Since this task is hardwired in the old release pipelines, it is pretty difficult to work around.

## Possible solution

The task currently uses a very old azure-devops-node-api@7.2.0 npm package.

https://github.com/microsoft/azure-pipelines-tasks/blob/5eafac932b229cf7efd8faba157241ec00821f1c/Tasks/DownloadBuildArtifactsV0/package.json#L23

The library implemented full support for `no_proxy` in late 2018, after the currently employed version was released: https://github.com/microsoft/azure-devops-node-api/issues/183. Upgrading to azure-devops-node-api@11.1.1 could solve the issue.

## Related Issues

- [x] #12135
- [x] #14812

## Task logs

```
2022-05-05T13:46:39.8732092Z ##[section]Starten: Artefakt herunterladen – _drop
2022-05-05T13:46:39.9410981Z ==============================================================================
2022-05-05T13:46:39.9411546Z Task         : Buildartefakte herunterladen
2022-05-05T13:46:39.9411964Z Description  : Hiermit werden Dateien heruntergeladen, die als Artefakte eines abgeschlossenen Builds gespeichert wurden.
2022-05-05T13:46:39.9412248Z Version      : 0.178.0
2022-05-05T13:46:39.9412527Z Author       : Microsoft Corporation
2022-05-05T13:46:39.9412897Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/download-build-artifacts
2022-05-05T13:46:39.9413205Z ==============================================================================
2022-05-05T13:46:41.5244641Z Fehler in "getBuildApi", Vorgang wird wiederholt. Ausstehende Versuche: 4
2022-05-05T13:47:25.8368410Z Fehler in "getBuildApi", Vorgang wird wiederholt. Ausstehende Versuche: 3
2022-05-05T13:49:40.9458432Z Fehler in "getBuildApi", Vorgang wird wiederholt. Ausstehende Versuche: 2
2022-05-05T13:55:41.0710960Z Fehler in "getBuildApi", Vorgang wird wiederholt. Ausstehende Versuche: 1
2022-05-05T14:01:41.1847885Z ##[error]Fehler in "getBuildApi": Error: unable to verify the first certificate
2022-05-05T14:01:41.2631560Z ##[error]Error: unable to verify the first certificate
2022-05-05T14:01:41.2659533Z ##[section]Abschließen: Artefakt herunterladen – _drop
```

I have verified that the error is only due to the traffic being routed through the proxy, not due to some actual certificate error. (We had the same issue in other tasks, ie. #16279.)

**Comment by Q-er** [comment]
I have the same problem.
Unfortunately your described solution/workaround did not work for me.
It seems like `azure-devops-node-api` >= `8.0.0` does not support DevOps Server 2020.
So we are stuck with `7.2.0`.

Instead what DID work for me was updating the `typed-rest-client` package inside of ` azure-devops-node-api` to the latest version.
I think this is the PR that made it work:
https://github.com/microsoft/typed-rest-client/pull/160

I also updated the `tunnel` dependency to version `0.0.6` because of this:
https://github.com/microsoft/azure-devops-node-api/issues/331

Basically what i did is:
```
cd node_modules/azure-devops-node-api/
npm install typed-rest-client@1.8.6 --save
npm install tunnel@0.0.6 --save
```

_Note: I am using Agent 2.202.0 on Linux, but this may also apply to the windows version._

**Comment by hangy** [comment]
> I have the same problem. Unfortunately your described solution/workaround did not work for me. It seems like `azure-devops-node-api` >= `8.0.0` does not support DevOps Server 2020. So we are stuck with `7.2.0`.

I actually never tried it, and only based my assumption on the updated library, which incorporated the fix. Did you just replace the DownloadBuildArtifacts@0 task server-side?
